packages/block-editor/src/components/block-preview/index.js

```diff
@@ -55,9 +55,17 @@ function ScaledBlockPreview( { blocks, viewportWidth } ) {
 
 				// Hack: we need  to reset the scaled elements margins
 				previewElement.style.marginTop = '0';
+
+				onReady( {
+					scale,
+					position: { x: offsetX * scale, y: offsetY },
+					ref: previewRef,
+				} );
 			} else {
 				const containerElementRect = containerElement.getBoundingClientRect();
-				setPreviewScale( containerElementRect.width / viewportWidth );
+				const scale = containerElementRect.width / viewportWidth;
+				setPreviewScale( scale );
+				onReady( { scale, ref: previewRef } );
```

Comment:
I'm not sure if we should be passing refs via an event handler it seems to be uncommon, can't we use the normal way of passing references in React? e.g., in this case, BlockPreview may have an optional __experimental ScaledPreviewRef property that is passed to ScaledBlockPreview component which then forwards the ref.

Reply:
I'm not understanding this feedback, Jorge :-/

> can't we use the normal way of passing references in React?

The reference is gotten just here, in this component, and the callback passes the reference as an argument as a chance to get keep working in the preview process if it's needed.

Reply:
Hi @retrofox I'm suggesting something identical to this commit https://github.com/WordPress/gutenberg/commit/e7aaee96f67eb7f5a27f16f8527f9f989003a92b where we allow the possibility of the BlockPreview user to pass its own reference, so the event does not need to pass it.

Reply:
Sounds Good 👌 Let me update the PR.

Reply:
Sorry, I think I completely misunderstood the conversation here:

> I'm not sure if we should be passing refs via an event handler

it isn't passing the reference. What is does is to give the reference as an argument in the callback function.

What the component does is adding a reference to be able to compute sizes in order to adjust the scale factor to apply to the preview block.

Removing the commit.